core/src/schema/namespace.rs

```diff
@@ -176,3 +254,137 @@ impl Compile for Namespace {
         Ok(Graph::Object(object_node))
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    #[test]
+    fn test_sort_simple() {
+        let mut namespace = Namespace {
+            collections: BTreeMap::new(),
+        };
+        let ref1: FieldRef = "visitors.address.postcode".parse().unwrap();
+        let ref2: FieldRef = "daughters.address.postcode".parse().unwrap();
+        //            let ref3: FieldRef = "winners.address.postcode".parse().unwrap();
+
+        namespace
+            .put_collection(
+                &"users".parse::<Name>().unwrap(),
+                Content::SameAs(crate::schema::SameAsContent { ref_: ref1 }),
+            )
+            .unwrap();
+        namespace
+            .put_collection(
+                &"sons".parse::<Name>().unwrap(),
+                Content::SameAs(crate::schema::SameAsContent { ref_: ref2 }),
+            )
+            .unwrap();
+        // namespace.put_collection(
+        //     &"losers".parse::<Name>().unwrap(),
+        //     Content::SameAs(crate::schema::SameAsContent { ref_: ref3 }),
+        // );
+        println!("sorted: {:?}", namespace.topo_sort());
+        assert!(
```

Comment:
The order does matter, I think. But there are only four acceptable possibilities in this case. The first two may come in any order, and the last two also. But they can not all mix.

Reply:
Ah, I see. Then it would likely make sense to define a test helper method that checks for each prefix of the list (so `for i in 0..list.len { check(&list[..i] }`) that the last element only depends on elements that come before. We could then use this helper for all tests instead of writing out the allowable orderings.

Reply:
Yes. Perfect. I'll do this